library/src/main/java/com/google/android/material/motion/runtime/TargetScope.java

```diff
@@ -85,6 +86,21 @@ void commitPlan(PlanInfo plan) {
     }
   }
 
+  void commitAddNamedPlan(PlanInfo planInfo) {
+    // remove first
+    Performer performer = getNamedPerformer(planInfo);
+    removeNamedPlan(planInfo, performer);
+    // then add
+    if (performer instanceof Performer.NamedPlanPerformance) {
```

Comment:
Perhaps we want to throw an assertion here if the performer doesn't implement named plan support?